include/zephyr/sys/mem_manage.h

```diff
@@ -194,17 +194,21 @@ static inline uintptr_t z_mem_phys_addr(void *virt)
 #if CONFIG_KERNEL_VM_BASE != 0
 		 (addr >= CONFIG_KERNEL_VM_BASE) &&
 #endif
-		 (addr < (CONFIG_KERNEL_VM_BASE +
-			  (CONFIG_KERNEL_VM_SIZE))),
```

Comment:
When both `CONFIG_KERNEL_VM_BASE` and `CONFIG_KERNEL_VM_SIZE` are 0, `addr` will never be lesser than 0 and therefore it would be essentially:

```c
__ASSERT(false, ...);